app/src/pages/explorer/dataset/DatasetGallery.vue

```diff
@@ -0,0 +1,271 @@
+import Dialog from '@/components/Dialog.vue'
+import { mapGetters, mapActions, mapMutations } from 'vuex'
+import reducer from '@/mixins/reduce'
+import optionalChainingUtil from '@/mixins/optional-chaining-util'
+import explorerQueryParams from '@/mixins/explorerQueryParams'
+
+export default {
+  name: 'viz-grid',
+  mixins: [reducer, explorerQueryParams, optionalChainingUtil],
+  data () {
+    return {
+      loading: true,
+      pageNumber: 1,
+      baseUrl: `${window.location.origin}/api/files/`,
+      dialog: {
+        title: ''
+      },
+      dialogLoading: false,
+      filter: '',
+      searchWord: '',
+      searchEnabled: false
+    }
+  },
+  components: {
+    pagination,
+    spinner,
+    dialogbox: Dialog
+  },
+  computed: {
+    ...mapGetters({
+      dialogBoxActive: 'dialogBox',
+      isAuth: 'auth/isAuthenticated',
+      isAdmin: 'auth/isAdmin',
+      items: 'explorer/sddDatasets/getAllDatasets',
+      page: 'explorer/sddDatasets/getPage',
+      total: 'explorer/sddDatasets/getTotal',
+      totalPages: 'explorer/sddDatasets/getTotalPages',
+      queryTimeMillis: 'explorer/sddDatasets/getQueryTimeMillis'
+    })
+  },
+  methods: {
+    ...mapActions('explorer/sddDatasets', ['loadItems']),
+    ...mapMutations({ toggleDialogBox: 'setDialogBox' }),
+    renderDialog (title, type, result, minWidth) {
+      this.dialog = {
+        title,
+        type,
+        minWidth,
+        dataset: result
+      }
+      this.toggleDialogBox()
+    },
+    async submitSearch () {
+      this.searchEnabled = !!this.searchWord // || !!this.filtersActive
+      this.pageNumber = 1
+      return await this.updateParamsAndCall(true)
+    },
+    async customReset (type) {
+      this.filter = ''
+      this.searchWord = null
+      await this.resetSearch(type)
+    },
+    async deleteDataset (dataset) {
+      if (!this.isAdmin) return // temporary safeguard
+      this.dialogLoading = true
+      await this.$store.dispatch('explorer/curation/deleteEntityNanopub', dataset.identifier)
+      await this.$store.dispatch('explorer/curation/deleteEntityES', {
+        identifier: dataset.identifier,
+        type: 'datasets'
+      })
+      this.toggleDialogBox()
+      this.dialogLoading = false
+      await this.loadItems()
+    },
+    editDataset (dataset) {
+      return this.$router.push(`/explorer/curate/sdd/edit/${this.getDatasetId(dataset)}`)
+    },
+    downloadFiles (item) {
+      if (item.distribution) {
+        fetch(item.distribution)
+      }
+    },
+    async localSearchMethod () {
+      if (this.searchEnabled) {
+        this.$store.dispatch('explorer/sddDatasets/searchDatasetKeyword', {
+          searchTerm: this.searchWord,
+          page: this.pageNumber
+        })
+      } else await this.loadItems(this.pageNumber)
+      this.loading = false
+    },
+    async loadItems (page = 1) {
+      this.loading = true
+      await this.$store.dispatch('explorer/sddDatasets/loadDatasets', { page })
```

Comment:
When action throws an error, that error either isn't caught and leaves the page stuck loading without any feedback to the user

Reply:
Good catch, updated this in most recent commit. I was following the pattern of Gallery.vue for this, so we may need to check that errors are caught in that file correctly as well.

Reply:
Sure